debezium-core/src/main/java/io/debezium/util/Loggings.java

```diff
@@ -72,4 +72,34 @@ public static void logErrorAndTraceRecord(Logger logger, Object record, String m
         logger.error(message, t);
         LOGGER.trace("Source of error is record '{}'", record);
     }
+
+    /**
+     * Log an info message and explicitly append the source of the info entry as a separate log
+     * entry that uses trace logging to prevent unintended leaking of sensitive data.
+     *
+     * @param logger the logger instance
+     * @param record the record the log entry is based upon
+     * @param message the info message to be logged
+     * @param arguments the arguments passed to the info message
+     */
+    public static void logInfoAndTraceRecord(Logger logger, Object record, String message,
+                                             Object... arguments) {
+        logger.info(message, arguments);
+        LOGGER.trace("Source of info is record '{}'", record);
+    }
+
+    /**
+     * Log trace message and explicitly append the source of the trace entry as a separate log
+     * entry that uses trace logging to centralise all record loggings.
+     *
+     * @param logger the logger instance
+     * @param record the record the log entry is based upon
+     * @param message the trace message to be logged
+     * @param arguments the arguments passed to the trace message
+     */
+    public static void logTraceAndTraceRecord(Logger logger, Object record, String message,
+                                              Object... arguments) {
+        logger.trace(message, arguments);
+        LOGGER.trace("Source of trace is record '{}'", record);
```

Comment:
I really don't think for use cases where we have an existing TRACE log operation that we should convert a single log entry into two log entries. That to me creates more noise than necessary in the logs.

Reply:
Hey @Naros
This has been done to ensure we only log sensitive content using Loggings class. The problem as described in the [JIRA](https://issues.redhat.com/browse/DBZ-8525) is that we have a risk of exposing sensitive data when TRACE logging is enabled as multiple classes currently log sensitive content. With this change, we aim to limit the exposure of sensitive data from Loggings class only. If anyone wants to redact sensitive content, they would only have to care about this single class and not scan the entire core for sensitive logs.

Reply:
@jpechane @Naros Some time back if I recall correctly that enabling TRACE log level was meant to expose sensitive data on purpose. How do I get this back? I didn't see any config added to enable it?

Reply:
Hi @b-goyal I see your point. I've always approached this a bit differently, where I see it more broadly that you use the non-TRACE levels to log non-sensitive information and leave TRACE specifically for that purpose. This is why I questions the point behind two trace log entries from different loggers as being a bit over-engineered in PoV.  But if you're looking for a way to tune the exclusion of sensitive data by log class, does make sense.

@rk3rn3r you still have it that way. What @b-goyal has done is allow the environment to be configured with TRACE at the root or within the `io.debezium` package but they can then override `io.debezium.util.Loggings` to use INFO or DEBUG to prevent the output of sensitive data, while retaining everything else.